packages/calypso-e2e/src/data-helper.ts

```diff
@@ -139,6 +139,24 @@ export function getCalypsoURL(
 	return url.toString();
 }
 
+/**
+ * Assembles and returns a URL to a specific page in WP-Admin.
+ *
+ * @param {string} siteSlug Slug of the site, typically wordpress or wpcomstaging.com.
+ * @param {string} path Path to the WP-Admin page.
+ * @returns {string} The full URL path.
+ * @throws {Error} If siteSlug is empty.
+ */
+export function getWPAdminURL( siteSlug: string, path: string ): string {
+	if ( ! siteSlug ) {
+		throw new Error( 'siteSlug must not be empty.' );
+	}
+	if ( siteSlug.match( /^(http:\/\/|https:\/\/)/ ) ) {
+		siteSlug = siteSlug.replace( /^(http:\/\/|https:\/\/)/, '' );
+	}
+	return new URL( path, `https://${ siteSlug }` ).href;
+}
+
```

Comment:
**Question/suggestion (non-blocking)**: Could you explain the purpose of this function a little bit more?

I think after reading the name/signature, I was expecting it to do a bit more than it does, which is really just safely join a path and hostname. I think that is something that could be safely offloaded to the caller as they would know if they are using a protocol already, and could just join the path the site host name themselves.

I wonder if a better abstraction would be to move this function to the `TestAccount` class? And then it could handle fetching the site name for you, and always set `protocol: true`.

And since it's called `getWPAdminURL` -- should the user have to specify the `wp-admin` path themselves? Or would it be better to have the function handle creating that part of the path? Otherwise we could just rename this to something like `getSiteUrl` since it could then be used for both `wp-admin` paths and non-admin paths!

Reply:
The intention of `getWPAdminURL` is to be a counterpart to the `getCalypsoURL` that we use almost everywhere. 

> I think that is something that could be safely offloaded to the caller as they would know if they are using a protocol already, and could just join the path the site host name themselves.

I'm a little less trusting of the caller. The way I see it is that if it's up to the caller how to assemble the WP-Admin URL, we will see a bunch of disparate ways it gets implemented. Not exact analogue, but we already see this with a Page POM's `visit` method; there are a few different signatures for the method[1].

> I wonder if a better abstraction would be to move this function to the TestAccount class? And then it could handle fetching the site name for you, and always set protocol: true.

There's precedent here with `TestAccount.getSiteURL` which in turn calls `DataHelper.getAccountSiteURL`. I am tepid on moving the `getCalypsoURL` and `getWPAdminURL` methods into the test account class though, because what if we want to construct URLs without being tied to a specific account? Having it in the DataHelper class keeps the function less coupled to the test accounts themselves.

[1] Examples:
- https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/pages/cart-checkout-page.ts#L97
- https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/pages/general-settings-page.ts#L41C52-L41C52
- https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/pages/plugins-page.ts#L81